### ISSUE TITLE
MatchXML issue when properties are reversed

### ISSUE THREAD

**Issue by arthurgustin** [onsi/gomega#234]
Hi, I don't know if my problem is a real one, but it doesn't work as I expect it to work.
Given the following ginkgo test:

```
It("should match in both case", func() {
    a := `<a foo="bar" ka="boom"></a>`
    b := `<a ka="boom" foo="bar"></a>`
    Expect(a).To(MatchXML(a))
    Expect(a).To(MatchXML(b))
})
```

It responds with:
```
Expected
      <a foo="bar" ka="boom"></a>
      to match XML of
      <a ka="boom" foo="bar"></a>
```

To me, both xml are equivalent, but correct me if I'm wrong (?)

In my program I don't have the control of the attributes order, so I end up with random unit test failures, which is not very convenient :/
@aubm

**Comment by arthurgustin** [comment]
It seems that DeepEqual needs slices to be in the same order ?
https://play.golang.org/p/AZQClnWwZM

**Comment by jvshahid** [comment]
fixed in 1eecca0ba8e6f5ea5a431ce21d3aee2af0b4c90b
